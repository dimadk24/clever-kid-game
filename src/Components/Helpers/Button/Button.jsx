import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

function Button({ children, className = '', ...other }) {
  return (
    <button type="button" className={`button ${className}`} {...other}>
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Button.defaultProps = {
  className: '',
};

export default Button;
