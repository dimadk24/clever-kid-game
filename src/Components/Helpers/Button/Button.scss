.button {
  border-style: solid;
  border-width: 1px;
  background-color: transparent;
  border-radius: 5px;
  padding: 5px 10px;
  transition-duration: 300ms;
}

.button:enabled {
  cursor: pointer;
  border-color: black;
}

.button:disabled {
  border-color: grey;
}

.button:hover:enabled,
.button:active:enabled {
  background-color: antiquewhite;
}
