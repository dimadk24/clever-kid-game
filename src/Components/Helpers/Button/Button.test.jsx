import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  it('should render button element', () => {
    const component = renderer.create(<Button>test</Button>);
    const tree = component.toJSON();
    expect(tree.type).toBe('button');
  });

  it('should match snapshot without classes', () => {
    const component = renderer.create(<Button>test</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with a class', () => {
    const component = renderer.create(
      <Button className="class">test</Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with 2 classes', () => {
    const component = renderer.create(
      <Button className="class another-class">test</Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain passed class', () => {
    const component = renderer.create(
      <Button className="class">test</Button>,
    );
    const tree = component.toJSON();
    expect(tree.props.className).toContain('class');
  });

  it('should contain 2 passed classes', () => {
    const component = renderer.create(
      <Button className="class another-class">test</Button>,
    );
    const tree = component.toJSON();
    expect(tree.props.className).toContain('class');
    expect(tree.props.className).toContain('another-class');
  });

  it('should have passed children', () => {
    const component = renderer.create(
      <Button>super test text</Button>,
    );
    const tree = component.toJSON();
    expect(tree.children).toHaveLength(1);
    expect(tree.children).toContain('super test text');
  });

  it('should pass all unknown props down to button', () => {
    const callback = jest.fn(() => ({ a: 1 }));
    const component = renderer.create(
      <Button onClick={callback}>test</Button>,
    );
    const tree = component.toJSON();
    expect(tree.props.onClick).toBe(callback);
  });
});
