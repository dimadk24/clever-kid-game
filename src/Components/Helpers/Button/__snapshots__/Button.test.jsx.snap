// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot with 2 classes 1`] = `
<button
  className="button class another-class"
  type="button"
>
  test
</button>
`;

exports[`Button should match snapshot with a class 1`] = `
<button
  className="button class"
  type="button"
>
  test
</button>
`;

exports[`Button should match snapshot without classes 1`] = `
<button
  className="button "
  type="button"
>
  test
</button>
`;
